### ISSUE TITLE
Integer ConfigItems always default to zero

### ISSUE THREAD

**Issue by ravron** [fastlane/fastlane#4336]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines] (https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md) 🔑

### Issue Description

`ConfigItem` incorrectly causes all instances with `data_type` `Integer` to default to `0`. 

The issue is the [`auto_convert_value` method on `ConfigItem`][1]. If the `data_type` is `Integer`, that method unconditionally casts `value` to `Integer`. Casting `nil` to `Integer` results in `0`, not `nil`. That means that by the time that control [reaches here][2], `value` is not `nil`, so the default value is not properly substituted.

A specific example of the bug is in `pilot`'s `wait_processing_interval` parameter. It [should default to `30`][3], but instead it is read as `0` [here, in the `BuildManager`][4]. As a result, the `sleep` has no effect, and instead just loops as fast as possible, constrained only by the response time of iTunes Connect, which is about a second in my tests. You can see the evidence of this issue in my log output, below: 

```
[12:05:25]: Waiting for iTunes Connect to process the new build
[12:05:27]: Waiting for iTunes Connect to process the new build
[12:05:28]: Waiting for iTunes Connect to process the new build
[12:05:29]: Waiting for iTunes Connect to process the new build
[12:05:30]: Waiting for iTunes Connect to process the new build
```

showing that we're hitting iTunes Connect about once per second, instead of the intended default once per thirty seconds. 

The fix should be pretty straightforward, and I may even take a swing at it after #3888. We'll want to check all `ConfigItem`s that use the type `Integer`, though, to make sure we're not introducing any regressions.

##### Complete output when running fastlane, including the stack trace and command used

```
▶ fastlane-test-app (master) $ fastlane beta
[12:03:55]: -------------------------------------------------
[12:03:55]: --- Step: Verifying required fastlane version ---
[12:03:55]: -------------------------------------------------
[12:03:55]: fastlane version valid
[12:03:55]: ------------------------------
[12:03:55]: --- Step: default_platform ---
[12:03:55]: ------------------------------
[12:03:55]: Driving the lane 'ios beta' 🚀
[12:03:55]: ------------------------------------
[12:03:55]: --- Step: increment_build_number ---
[12:03:55]: ------------------------------------
[12:03:55]: $ cd /Users/ravron/src/oss/fastlane-test-app && agvtool next-version -all
[12:03:56]: ▸ Setting version of project fastlane-test-app to:
[12:03:56]: ▸ 7.
[12:03:56]: ▸ Also setting CFBundleVersion key (assuming it exists)
[12:03:56]: ▸ Updating CFBundleVersion in Info.plist(s)...
[12:03:56]: ▸ Updated CFBundleVersion in "fastlane-test-app.xcodeproj/../fastlane-test-app/Info.plist" to 7
[12:03:56]: ▸ Updated CFBundleVersion in "fastlane-test-app.xcodeproj/../fastlane-test-appTests/Info.plist" to 7
[12:03:56]: ▸ Updated CFBundleVersion in "fastlane-test-app.xcodeproj/../fastlane-test-appUITests/Info.plist" to 7
[12:03:56]: ------------------------
[12:03:56]: --- Step: git_commit ---
[12:03:56]: ------------------------
[12:03:56]: $ git commit -m 'Bump build number' .
[12:03:56]: ▸ [master 22c80ca] Bump build number
[12:03:56]: ▸ 4 files changed, 5 insertions(+), 5 deletions(-)
[12:03:56]: Successfully committed "." 💾.
[12:03:56]: -----------------
[12:03:56]: --- Step: gym ---
[12:03:56]: -----------------
[12:03:56]: Installing provisioning profile...
[12:03:56]: xcrun xcodebuild -list -project './fastlane-test-app.xcodeproj'

+---------------------------+----------------------------------------------------------------------+
|                                      Summary for gym 1.6.2                                       |
+---------------------------+----------------------------------------------------------------------+
| scheme                    | fastlane-test-app                                                    |
| project                   | ./fastlane-test-app.xcodeproj                                        |
| provisioning_profile_path | /Users/ravron/src/oss/fastlane-test-app/AppStore_com.ravron.fastl... |
| destination               | generic/platform=iOS                                                 |
| output_name               | fastlane-test-app                                                    |
| clean                     | false                                                                |
| output_directory          | .                                                                    |
| silent                    | false                                                                |
| use_legacy_build_api      | false                                                                |
| buildlog_path             | ~/Library/Logs/gym                                                   |
+---------------------------+----------------------------------------------------------------------+

[12:03:58]: $ set -o pipefail && xcodebuild -scheme 'fastlane-test-app' -project './fastlane-test-app.xcodeproj' -destination 'generic/platform=iOS' -archivePath '/Users/ravron/Library/Developer/Xcode/Archives/2016-04-23/fastlane-test-app 2016-04-23 12.03.58.xcarchive' archive | tee /Users/ravron/Library/Logs/gym/fastlane-test-app-fastlane-test-app.log | xcpretty
[12:03:59]: ▸ Building fastlane-test-app/fastlane-test-app [Release]
[12:03:59]: ▸ Check Dependencies
[12:04:00]: ▸ Compiling ViewController.swift
[12:04:00]: ▸ Compiling AppDelegate.swift
[12:04:00]: ▸ Compiling ViewController.swift
[12:04:00]: ▸ Compiling AppDelegate.swift
[12:04:00]: ▸ Compiling fastlane-test-app_vers.c
[12:04:00]: ▸ Compiling fastlane-test-app_vers.c
[12:04:00]: ▸ Linking fastlane-test-app
[12:04:00]: ▸ Linking fastlane-test-app
[12:04:01]: ▸ Compiling LaunchScreen.storyboard
[12:04:01]: ▸ Compiling Main.storyboard
[12:04:01]: ▸ Processing Info.plist
[12:04:01]: ▸ Generating 'fastlane-test-app.app.dSYM'
[12:04:02]: ▸ Touching fastlane-test-app.app
[12:04:04]: ▸ Signing /Users/ravron/Library/Developer/Xcode/DerivedData/fastlane-test-app-aaesqqfcggjaxuehimezpfbmtnaj/Build/Intermediates/ArchiveIntermediates/fastlane-test-app/InstallationBuildProductsLocation/Applications/fastlane-test-app.app
[12:04:04]: ▸ Touching fastlane-test-app.app.dSYM
[12:04:04]: ▸ Archive Succeeded
[12:04:04]: Successfully stored the archive. You can find it in the Xcode Organizer.
[12:04:04]: $ /usr/bin/xcrun /Library/Ruby/Gems/2.0.0/gems/gym-1.6.2/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive -exportOptionsPlist '/var/folders/g7/rfpy5fcx4k9_vx_v7yqpkpy5wz1tvr/T/gym20160423-58703-i1y9fb_config.plist' -archivePath '/Users/ravron/Library/Developer/Xcode/Archives/2016-04-23/fastlane-test-app 2016-04-23 12.03.58.xcarchive' -exportPath '/var/folders/g7/rfpy5fcx4k9_vx_v7yqpkpy5wz1tvr/T/gym20160423-58703-1jpw430.gym_output' 
[12:04:17]: Compressing 1 dSYM(s)
[12:04:17]: $ cd '/Users/ravron/Library/Developer/Xcode/Archives/2016-04-23/fastlane-test-app 2016-04-23 12.03.58.xcarchive/dSYMs' && zip -r '/Users/ravron/src/oss/fastlane-test-app/fastlane-test-app.app.dSYM.zip' *.dSYM
[12:04:17]: ▸ updating: fastlane-test-app.app.dSYM/ (stored 0%)
[12:04:17]: ▸ updating: fastlane-test-app.app.dSYM/Contents/ (stored 0%)
[12:04:17]: ▸ updating: fastlane-test-app.app.dSYM/Contents/Info.plist (deflated 52%)
[12:04:17]: ▸ updating: fastlane-test-app.app.dSYM/Contents/Resources/ (stored 0%)
[12:04:17]: ▸ updating: fastlane-test-app.app.dSYM/Contents/Resources/DWARF/ (stored 0%)
[12:04:17]: ▸ updating: fastlane-test-app.app.dSYM/Contents/Resources/DWARF/fastlane-test-app (deflated 73%)

[12:04:17]: Successfully exported and compressed dSYM file
[12:04:17]: Successfully exported and signed the ipa file:
[12:04:17]: /Users/ravron/src/oss/fastlane-test-app/fastlane-test-app.ipa
[12:04:17]: -------------------
[12:04:17]: --- Step: pilot ---
[12:04:17]: -------------------
[12:04:17]: Login to iTunes Connect (nerylix@me.com)
[12:04:20]: Login successful
[12:04:20]: Ready to upload new build to TestFlight (App: 1107292001)...
[12:04:20]: Going to upload updated app to iTunes Connect
[12:04:20]: This might take a few minutes, please don't interrupt the script
[12:05:25]: iTunes Transporter successfully finished its job
[12:05:25]: ------------------------------------------------------------------------------------------------------
[12:05:25]: Successfully uploaded package to iTunes Connect. It might take a few minutes until it's visible online.
[12:05:25]: ------------------------------------------------------------------------------------------------------
[12:05:25]: Successfully uploaded the new binary to iTunes Connect
[12:05:25]: Waiting for iTunes Connect to process the new build
[12:05:27]: Waiting for iTunes Connect to process the new build
[12:05:28]: Waiting for iTunes Connect to process the new build
[12:05:29]: Waiting for iTunes Connect to process the new build
[12:05:30]: Waiting for iTunes Connect to process the new build
<snip>
[12:15:52]: Waiting for iTunes Connect to process the new build
[12:15:53]: Waiting for iTunes Connect to process the new build
[12:15:53]: Waiting for iTunes Connect to process the new build
[12:15:54]: Waiting for iTunes Connect to process the new build
[12:15:55]: Waiting for iTunes Connect to process the new build
[12:15:58]: Waiting for iTunes Connect to finish processing the new build (1.0 - 7)
[12:15:58]: Successfully finished processing the build
[12:15:58]: You can now tweet: 
[12:15:58]: iTunes Connect #iosprocessingtime 11 minutes
[12:15:58]: Distributing new build to testers
[12:16:00]: Successfully distributed build to beta testers 🚀

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | increment_build_number              | 0           |
| 4    | git_commit                          | 0           |
| 5    | gym                                 | 20          |
| 6    | pilot                               | 703         |
+------+-------------------------------------+-------------+

[12:16:00]: fastlane.tools just saved you 12 minutes! 🎉
```

### Configuration Files

Fastfile:
```ruby
▶ fastlane (master) $ cat Fastfile 
# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.81.0"

default_platform :ios

platform :ios do
  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    increment_build_number
    git_commit(path:".", message:"Bump build number")
    gym(scheme: "fastlane-test-app") # Build your app - more options available
    # Note that wait_processing_interval is not specified here, so it *should* default to 30
    pilot
  end
end
```

### Environment

fastlane version: `1.81.0`

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? No

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? No

[1]: https://github.com/fastlane/fastlane/blob/fa5fc3ef7ef22c9e6b760f0e15b7f2498beb151e/fastlane_core/lib/fastlane_core/configuration/config_item.rb#L93
[2]: https://github.com/fastlane/fastlane/blob/fa5fc3ef7ef22c9e6b760f0e15b7f2498beb151e/fastlane_core/lib/fastlane_core/configuration/configuration.rb#L180
[3]: https://github.com/fastlane/fastlane/blob/fa5fc3ef7ef22c9e6b760f0e15b7f2498beb151e/pilot/lib/pilot/options.rb#L82
[4]: https://github.com/fastlane/fastlane/blob/master/pilot/lib/pilot/build_manager.rb#L72

**Comment by TKBurner** [comment]
@ravron Thanks for reaching out about this and including such detailed investigation. We are going to investigate this over here. But, if anyone wants to submit a PR we can have preliminary discussions about that here too. Thanks! 👍

**Comment by ravron** [comment]
@TKBurner No problem. A note: though this issue affects pilot, it's really more of fastlane_core bug, so that label might be more appropriate. Of course, I defer to your judgement.

**Comment by TKBurner** [comment]
@ravron Thanks for following up. I'll mark it that way 👍

**Comment by asfalcone** [comment]
This will go out in the next release of `fastlane_core`
